CONTRIBUTING.md

```diff
@@ -1,6 +1,132 @@
+
+### Standard output, warnings, and errors
+
+- **Print Statement:** Used only for outputs in methods and scenarios explicitly requested by the user
+- **Errors and Exceptions:**  Always use the `raise` with built-in or custom exception classes. The nondescript `Exception` class should be avoided as much as possible, in favor of more specific exceptions (*IOError, ValueError* etc.).
+- **Warnings:**  Always use the `warnings.warn(message, warning_class)`for warnings. These get redirected to `log.warning()` by default, but one can still use the standard warning-catching mechanism and custom warning classes.
+- **Debugging and  Informational messages:**  Always use `log.info(message)` and `log.debug(message)`. The logging system uses the built-in Python logging module.
+
+### Data and Configuration
+
+- **Storing Data:**
+    - Packages can include data in a directory named *data* inside a subpackage source directory as long as it is less than about 100 kB.
+    - If the data exceeds this size, it should be hosted outside the source code repository, either at a third-party location on the internet.
+
+### Documentation and Testing
+
+- **Docstrings:**
+    - Docstrings must be provided for all public classes, methods, and functions.
+    - Docstrings should follow the [numpydoc style](https://numpydoc.readthedocs.io/en/latest/format.html) and reStructured Text format.
+    - Write usage examples in the docstrings of all classes and functions whenever possible. These examples should be short and simple to reproduce. Users should be able to copy them verbatim and run them.
+- **Unit tests:**  Provided for as many public methods and functions as possible, and should adhere to the standards set in the Testing Guidelines.
+- **Building Documentation:**
+    - Use sphinx to build the documentation.
+    - All extra documentation should go into a /docs sub-directory under the main stingray directory.
+
+## Testing Guidelines
+
+---
+
+The testing framework used by stingray is the pytest framework, it is accessed through the python [setup.py](http://setup.py/) test command. To run the tests, you will need to make sure you have the pytest package (version 3.1 or later) installed.
+
+- All tests should be py.test compliant: [http://pytest.org/latest/](http://pytest.org/latest/).
+- Keep all tests in a /tests subdirectory under the main stingray directory.
+- Write one test script per module in the package.
+- Extra examples can go into an /examples folder in the main stingray directory, scripts that gather various data analysis tasks into longer procedures into a /scripts folder in the same location.
+
+## Community Guidelines
```

Comment:
You can just point to the CODE_OF_CONDUCT here

Reply:
Changed